### ISSUE TITLE
Fix `font-family-no-missing-generic-family-keyword` false positives for `font-family` descriptor in `@font-palette-values`

### ISSUE THREAD

**Issue by grommasdietz** [stylelint/stylelint#8139]
### What minimal example or steps are needed to reproduce the bug?

```css
@font-palette-values --example {
  font-family: example;
  override-colors: 0 red;
}

```

### What minimal configuration is needed to reproduce the bug?

```json
"rules": {
  "font-family-no-missing-generic-family-keyword": true
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAzCA7ALgWgA4EMAbGTTGbANyIFcYBnAAm2xgA98BbXYh4AHXQMGaLNhScAloQCe8Bm07cYAbgFCIFGACctE2NkiEIWunIAMDLTCir0AXxAAaECikwAcpziIFXYgDowOjoncAxXAHMEEH50Pkw+EC1qYjpEuVj4rJcMHHEOKWlsdAhsAuCJdAjsCJh0bQkwMUkZbABrGGkAd2ModIZMZJgBeLsBMfRQyHRIgDFjDnxMaIArOgxQ2FwQxFihRLpMaWJCSoSEBkTCJfpz8ZA7IA)

### Which Stylelint-related dependencies are you using?

```json
  "stylelint": "^16.10.0",
  "stylelint-config-standard-scss": "^13.1.0",
```

### What did you expect to happen?

`font-family-no-missing-generic-family-keyword` should ignore the absence of generic font-family keyword when defining font-palette-values, since the color palette definition is only valid when used for color font. 

### What actually happened?

The following problem was reported:

```
2:16-23
ERROR Unexpected missing generic font family (font-family-no-missing-generic-family-keyword)
```

**Comment by jeddy3** [comment]
@grommasdietz Thanks for the report, demo and for using the template.

The rule should ignore [`font-family` descriptors](https://drafts.csswg.org/css-fonts/#font-family-2-desc) inside `@font-palette-values`.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.
